packages/atomic/src/components/commerce/atomic-product-multi-value-text/atomic-product-multi-value-text.ts

```diff
@@ -0,0 +1,213 @@
+ */
+@customElement('atomic-product-multi-value-text')
+@bindings()
+@withTailwindStyles
+export class AtomicProductMultiValueText
+  extends LitElement
+  implements InitializableComponent<CommerceBindings>
+{
+  static styles = [unsafeCSS(styles)];
+
+  @state()
+  bindings!: CommerceBindings;
+
+  @state()
+  error!: Error;
+
+  public breadcrumbManager!: BreadcrumbManager;
+
+  private productController = createProductContextController(this);
+
+  /**
+   * The field that the component should use.
+   * The component will try to find this field in the `Product.additionalFields` object unless it finds it in the `Product` object first.
+   * Make sure this field is present in the `fieldsToInclude` property of the `atomic-commerce-interface` component.
+   */
+  @property({reflect: true})
+  public field!: string;
+
+  /**
+   * The maximum number of field values to display.
+   * If there are _n_ more values than the specified maximum, the last displayed value will be "_n_ more...".
+   */
+  @property({reflect: true, type: Number, attribute: 'max-values-to-display'})
+  public maxValuesToDisplay = 3;
+
+  /**
+   * The delimiter used to separate values when the field isn't indexed as a multi value field.
+   */
+  @property({reflect: true})
+  public delimiter: string | null = null;
+
+  @state() private values: string[] = [];
+
+  public initialize() {
+    let searchOrListing: ProductListing | Search;
+
+    if (this.bindings.interfaceElement.type === 'product-listing') {
+      searchOrListing = buildProductListing(this.bindings.engine);
+    } else {
+      searchOrListing = buildSearch(this.bindings.engine);
+    }
+
+    this.breadcrumbManager = searchOrListing.breadcrumbManager();
+
+    this.values = this.initializeValues();
+  }
+
+  private initializeValues(): string[] {
+    const product = this.productController.item;
+    if (!product) {
+      return [];
+    }
+    const property = ProductTemplatesHelpers.getProductProperty(
+      product,
+      this.field
+    );
+    if (property === null) {
+      return [];
+    }
+    if (Array.isArray(property)) {
+      return property.map((v) => `${v}`.trim());
+    }
+
+    if (typeof property !== 'string' || property.trim() === '') {
+      this.error = new Error(
+        `Could not parse "${property}" from field "${this.field}" as a string array.`
+      );
+      return [];
+    }
+
+    return this.delimiter
+      ? property.split(this.delimiter).map((value) => value.trim())
+      : [property];
+  }
+
+  private get facetSelectedValues() {
+    return this.breadcrumbManager.state.facetBreadcrumbs
+      .filter((facet) => facet.field === this.field)
+      .reduce((values, facet) => {
+        return values.concat(
+          facet.values.map(({value}) => (value as RegularFacetValue).value)
+        );
+      }, [] as string[]);
```

Comment:
The array is iterated twice over; instead, one should fold the filter within the reduce.
```suggestion
    return this.breadcrumbManager.state.facetBreadcrumbs
      .reduce((values, facet) => {
        if(facet.field === this.field) {
          return values;
        }
        return values.concat(
          facet.values.map(({value}) => (value as RegularFacetValue).value)
        );
      }, [] as string[]);
```

Reply:
https://github.com/coveo/ui-kit/pull/4224/files#r1793766243 😆